Brokerages/Zerodha/ZerodhaBrokerage.cs

```diff
@@ -0,0 +1,1352 @@
+                var end = historyRequest.EndTimeUtc.ConvertTo(DateTimeZone.Utc, TimeZones.Kolkata);
+
+                var history = Enumerable.Empty<BaseData>();
+
+                if (Config.GetBool("zerodha-history-subscription"))
+                {
+                    switch (historyRequest.Resolution)
+                    {
+                        case Resolution.Minute:
+                            history = GetHistoryForPeriod(historyRequest.Symbol, start, end, historyRequest.Resolution, "minute");
+                            break;
+
+                        case Resolution.Hour:
+                            history = GetHistoryForPeriod(historyRequest.Symbol, start, end, historyRequest.Resolution, "60minute");
+                            break;
+
+                        case Resolution.Daily:
+                            history = GetHistoryForPeriod(historyRequest.Symbol, start, end, historyRequest.Resolution, "day");
+                            break;
+                    }
+                }
+
+                foreach (var baseData in history)
+                {
+                    yield return baseData;
+                }
+            }
+        }
+
+        private IEnumerable<BaseData> GetHistoryForPeriod(Symbol symbol, DateTime start, DateTime end, Resolution resolution, string zerodhaResolution)
+        {
+            Log.Debug("ZerodhaBrokerage.GetHistoryForPeriod();");
+            var scripSymbol = _symbolMapper.GetZerodhaInstrumentToken(symbol.Value, symbol.ID.Market);
+            var candles = _kite.GetHistoricalData(scripSymbol.ToStringInvariant(), start, end, zerodhaResolution);
+
+            if (!candles.Any())
+            {
+                OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Warning, "NoHistoricalData",
+                    $"Exchange returned no data for {symbol} on history request " +
+                    $"from {start:s} to {end:s}"));
+            }
+
+            var period = resolution.ToTimeSpan();
+
+            foreach (var candle in candles)
+            {
+                yield return new TradeBar()
+                {
+                    Time = candle.TimeStamp.ConvertFromUtc(TimeZones.Kolkata),
+                    Symbol = symbol,
+                    Low = candle.Low,
+                    High = candle.High,
+                    Open = candle.Open,
+                    Close = candle.Close,
+                    Volume = candle.Volume,
+                    Value = candle.Close,
+                    DataType = MarketDataType.TradeBar,
+                    Period = period,
+                    EndTime = candle.TimeStamp.Add(period).ConvertFromUtc(TimeZones.Kolkata)
+                };
+            }
+        }
+
+        /// <summary>
+        /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
+        /// </summary>
+        public override void Dispose()
+        {
+        }
```

Comment:
missing to dispose of the `_aggregator.DisposeSafely()`, should it dispose of the WebSocket too?